src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -186,10 +186,9 @@ class NBDServer
 
     if (ret < 0) {
       ctx->reply.error = htonl(-ret);
-    } else if ((ctx->command == NBD_CMD_WRITE || ctx->command == NBD_CMD_READ)
-	       && ret != static_cast<int>(ctx->request.len)) {
+    } else if ((ctx->command == NBD_CMD_WRITE) && ret != 0) {
```

Comment:
@trociny I think check return value with 0 might be enough.  
I tried to add read return value check 
``
else if (ctx->command == NBD_CMD_READ && ret  != ctx->request.len)
``
When I mount nbd image  it will show cannot read superblock.

I need others help review it.